main.js

```diff
@@ -0,0 +1,86 @@
+const form = document.getElementById("formId");
+
+function formValidation() {
+
+    const inputSearchObject = document.getElementById("inputSearch"); //captura de los elementos del formulario
+    const inputEmailObject = document.getElementById("inputEmail");
+    const inputPwdObject1 = document.getElementById("inputPwd1");
+    const inputPwdObject2 = document.getElementById("inputPwd2");
+    const inputProvinceObject = document.getElementById("inputProvince");
+
+    deleteInvalid();                                //métodos de comprobación y limpieza de estado de los campos
+    verifySearch(inputSearchObject);
+    verifyEmail(inputEmailObject);
+    verifyPwd(inputPwdObject1, inputPwdObject2);
+    verifyProvince(inputProvinceObject);
+
+}
+
+
+document.getElementById("formId").addEventListener('submit', function (e) { //Prevent Default para prevenir el borrado del formulario al submit del formulario
+    formValidation();
+    e.preventDefault();
+});
+
+
+function deleteInvalid() { //elimina las clases valid / invalid de los campos
+    for (elemento of form) {
+        elemento.classList.remove('is-invalid');
+        elemento.classList.remove('is-valid');
+    }
+}
+
+function verifySearch(search) { //verifica el campo de busqueda , min 4 caracteres y no vacío
+    if (search.value.length < 4 || search.value == "") {
+        search.classList.add("is-invalid");
+        document.getElementById("errorSearch").textContent = "We need at least 4 carachters";
+    } else {
+        search.classList.add("is-valid");
+        document.getElementById("okSearch").textContent = "Ok";
+    }
+}
+
+function verifyEmail(email) { //verifica email
+    const regExp = /[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9-]+(?:\.[a-zA-Z0-9-]+)/gi;
+
+    if (regExp.test(email.value)) {
+        email.classList.add('is-valid');
+        document.getElementById('okEmail').textContent = 'Ok';
+    } else {
+        email.classList.add('is-invalid');
+        document.getElementById('errorEmail').textContent = 'Invalid email format';
+    }
+}
+
+function verifyPwd(pwd1, pwd2) { // verifica passwords , requisitos min. 8 carácteres , una mayúscula , una minúscula y un número
```

Comment:
Esta función tiene bastante código repetido. ¿No se puede evitar?

Reply:
He realizado los cambios con un array donde almaceno los dos objetos html e iterandolos con un bucle for.